picamera2/picamera2.py

```diff
@@ -44,6 +46,66 @@ class Preview(Enum):
     QTGL = 3
 
 
+class CameraManager:
+    def __init__(self):
+        self.cameras = {}
+        self.setup()
+        self._lock = threading.Lock()
```

Comment:
Looks just a bit strange to create the lock after `setup()` starts the thread that uses it. Or should we avoid calling `setup()` here because it will presumably happen when the first camera gets `add`ed?

Reply:
Good point, will avoid calling setup there